### ISSUE TITLE
Add support for original command-line options

### ISSUE THREAD

**Issue by dginev** [dginev/latexml-oxide#191]
These are best bookkept at the perl `LaTeXML::Common::Config` module, but are also described in each original executable in the perl project's `bin/` directory.

For Rust, it would be nice to get some of the more modern features, such as option auto-completion, via basing option support on a community library such as [clap](https://crates.io/crates/clap). There is an interesting [overview post](https://rust-cli-recommendations.sunshowers.io/cli-parser.html) on CLI parsers for Rust, as well as [a public benchmark](https://github.com/rosetta-rs/argparse-rosetta-rs) for some of the mainstream options..

**Comment by dginev** [comment]
## Resolution — clap adopted, real-feature options wired, remainder documented

The core request here — move to a modern parser library (clap was the concrete suggestion) — is **done**. `clap` 4 derive is the option front-end (autocompletion-ready, with Perl-name aliases such as `--destination`, `--noparse`, `--presentationmathml`, `--contentmathml`, `--xmath`, `--navtoc`, …).

Every `latexmlc` option whose engine feature already works end-to-end is now wired. The remaining options are **deliberately** not wired, for the reasons documented below, so this is being closed as "parser migrated + option-coverage swept." Per-flag work continues as each underlying feature lands — tracked in [`docs/ISSUE_AUDIT.md`](https://github.com/dginev/latexml-oxide/blob/main/docs/ISSUE_AUDIT.md) and [`docs/SYNC_STATUS.md`](https://github.com/dginev/latexml-oxide/blob/main/docs/SYNC_STATUS.md), not this meta-issue.

Landed in **#276** (Release preparation, July 10).

### Wired (feature exists → flag exposed)

`--strict`, `--includestyles`, `--comments`, `--xml`, `--embed`, `--nopost`, `--nosplit`, `--nopmml`/`--nocmml`/`--nomathtex`/`--noxmath`, `--navtoc`, `--timestamp`, `--icon`, `--nographicimages`/`--graphicimages`, `--numbersections`, `--mathparse`, `--invisibletimes`, `--defaultresources` — plus the pre-existing core surface (`--dest`/`--destination`, `--format`, `--stylesheet`, `--post`, `--pmml`/`--cmml`, `--keep-xmath`, `--mathtex`, `--nomathparse`, `--nobibtex`/`--bibtex`, `--whatsout`, `--source-map`, `--debug`, …).

### Not wired — and why

**A. Postponed to a near-term release (feature nearly there).**
- `--validate` / `--novalidate` — real RelaxNG validation is gated on the `rust-libxml` fork exposing a safe RelaxNG interface (planned `libxml 0.3.16`); `Post::Document::validate()` is a stub today. Plan in `SYNC_STATUS.md`.
- `--profile` (+ its `--mode` alias) — the biggest single remaining item; planned as **TOML** profiles deserialized into the clap option struct (design in `OXIDIZED_DESIGN.md` → Future Work), not Perl `.opt` files.

**B. Feature not implemented yet — flag kept a hard parse error (no accept-and-warn stub).**
- Image pipelines: `--mathimages` / `--mathsvg` / `--pictureimages` — need the currently-unwired LaTeXImages `latex`+`dvipng` pipeline.
- `--svg` — **deferred**: the HTML5 XSLT already renders `<ltx:picture>` as inline `<svg>` by default, so the standalone `svg.rs` post-processor is redundant and produces divergent, unverified output.
- Alternate output backends: `--jats`, `--html4`, `--tex`, `--box`.
- Cross-reference / bibliography / index post-phases: `--crossref`, `--bibliography`, `--index`, `--permutedindex`, `--splitbibliography`, `--prescan`, `--dbfile`, `--urlstyle`, `--base`, and the `--noscan` off-switch (Scan itself **is** wired as a post phase; only the disable knob is parked with this cluster).
- Math-encoding knobs: `--openmath`, `--unicodemath`, `--plane1`, `--hackplane1`, `--parallelmath`, `--linelength`, `--mathimagemagnification`.
- Daemon mode: `--port`, `--expire`, `--cache_key`, `--address`, `--autoflush`.
- `--omitdoctype` — DTD-only; the Rust port supports **RelaxNG only** (no DTD), so this is a permanent non-goal.

**C. Intentional non-goals.**
- `--output` — superseded by `--destination` / `--dest`.

### Policy

Deferred-feature flags are left as clap "unexpected argument" **errors**, not accept-and-warn stubs: a strict parser never silently accepts a flag whose feature is absent (no misleading success). Each is revisited when its feature lands. The parser-library question this issue raised is settled — **clap 4 derive**.

Closing as complete for the migration + coverage sweep.
